pkg/task/storage.go

```diff
@@ -0,0 +1,155 @@
+package task
+
+import (
+	"container/heap"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+)
+
+// TaskStorage is a leveldb-backed priority queue.
+// storage is persisted before the in-memory queue to prevent inconsistency between restarts; what
+// you see in the queue is the same as what is written the database
+type TaskStorage struct {
+	Max       int
```

Comment:
What would happen if the value of `Max` changes across restarts?

Reply:
If max increases, then more items will be allowed on. If max decreases, pushes would be denied until queue length is less than max.